api/db-cdr/generate-cdr/libproject/devstart.rb

```diff
@@ -140,27 +138,43 @@ def publish_cdr(cmd_name, args)
         #{table_match_filter} #{table_skip_filter}}
 
     # Delete the intermediate dataset.
-    common.run_inline %W{bq rm -rf --dataset #{ingest_dataset}}
+    common.run_inline %W{bq rm -r -f --dataset #{ingest_dataset}}
 
-    auth_domain_group = get_auth_domain_group(op.opts.project)
+    auth_domain_group_email = get_auth_domain_group_email(op.opts.project)
 
     config_file = Tempfile.new("#{op.opts.bq_dataset}-config.json")
     begin
       json = JSON.parse(
         common.capture_stdout %{bq show --format=prettyjson #{dest_dataset}})
       existing_groups = Set[]
+      existing_users = Set[]
       for entry in json["access"]
         if entry.key?("groupByEmail")
           existing_groups.add(entry["groupByEmail"])
         end
+        if entry.key?("userByEmail")
+          existing_users.add(entry["userByEmail"])
+        end
       end
-      if existing_groups.include?(auth_domain_group)
-        common.status "#{auth_domain_group} already in ACL, skipping..."
+
+      if existing_groups.include?(auth_domain_group_email)
+        common.status "#{auth_domain_group_email} already in ACL, skipping..."
       else
-        common.status "Adding #{auth_domain_group} as a READER..."
-        new_entry = { "groupByEmail" => auth_domain_group, "role" => "READER"}
+        common.status "Adding #{auth_domain_group_email} as a READER..."
+        new_entry = { "groupByEmail" => auth_domain_group_email, "role" => "READER"}
         json["access"].push(new_entry)
       end
+
+      # if the app SA's in too many groups, it won't gain READER transitively.
+      # add it directly, to make sure.
```

Comment:
nit: maybe include slack link in here for further context: https://pmi-engteam.slack.com/archives/CHRN2R51N/p1609869521078200?thread_ts=1609796171.063800&cid=CHRN2R51N